### PR TITLE
BUDI-6863 - Provide some functions to check the path in the context

### PR DESCRIPTION
**Pull request by jvcalderon**
## Description

Provides useful utility functions in order to check the route in context:

- `isServingBuilder`
- `isServingBuilderPreview`
- `isPublicApiRequest`

This function will be useful in following PRs, but they deliver value for themselves.

## Documentation
- [X] I have reviewed the budibase documentatation to verify if this feature requires any changes. If changes or new docs are required I have written them.

⚠️ **UPDATE THE SUBMODULE REFERENCE ONCE THIS PR WAS MERGED:** https://github.com/Budibase/budibase-pro/pull/182 ⚠️

### CODE REVIEW

**Review by shogunpurple — APPROVED** [review]
LGTM! Nice to have tests too.

File: packages/backend-core/src/utils/utils.ts [hunk 1]
```diff
@@ -69,6 +65,18 @@ export function isServingApp(ctx: Ctx) {
   return false
 }
 
+export function isServingBuilder(ctx: Ctx): boolean {
+  return ctx.path.startsWith(BUILDER_REFERER_PREFIX)
```
Comment (Rory-Powell) [inline packages/backend-core/src/utils/utils.ts:69]: Just to note some differences on the builder urls

- `/builder/app` - editing an app, including data, design, automation and any screens within
- `/builder/portal` - everywhere outside of an app, aside from authentication screens

We could possibly rename `BUILDER_REFERER_PREFIX` to `BUILDER_APP_PREFIX` to be more clear as the current constants looks very 'referrer' specific 


Background on the `BUILDER_REFERER_PREFIX` - it's only used for determining the `appId` in `getAppIdFromCtx`. This is needed when using the app builder as we don't have the app id anywhere else in the request at that point. We get it from the referrer of any requests sent when using the builder e.g. when fetching tables.

File: packages/backend-core/src/utils/tests/utils.spec.ts [hunk 2]
```diff
@@ -106,4 +108,79 @@ describe("utils", () => {
...
+    beforeEach(() => {
+      ctx = structures.koa.newContext()
+    })
+
+    it("returns true if current path is in builder", async () => {
+      ctx.path = "/builder/app/app_"
+      expectResult(true)
+    })
+
+    it("returns false if current path is not in builder", async () => {
```
Comment (Rory-Powell) [inline packages/backend-core/src/utils/tests/utils.spec.ts:127]: might be more clear:
```suggestion
    it("returns false if current path doesn't have '/' suffix", async () => {
```

**Review by Rory-Powell — APPROVED** [review]
Minor comments, otherwise LGTM!

File: packages/backend-core/src/utils/tests/utils.spec.ts [hunk 3]
```diff
@@ -5,6 +5,8 @@ import * as db from "../../db"
 import { Header } from "../../constants"
 import { newid } from "../../utils"
 import env from "../../environment"
+import { beforeEach } from "@jest/globals"
```
[the remark this answers is not in the document]
Comment (jvcalderon) [reply packages/backend-core/src/utils/tests/utils.spec.ts]: Done.

File: packages/backend-core/src/utils/tests/utils.spec.ts [hunk 4]
```diff
@@ -106,4 +108,79 @@ describe("utils", () => {
...
+      ctx.path = "/api/public/v1/invoices"
+      expectResult(true)
+
+      ctx.path = "/api/public/v1"
+      expectResult(true)
+    })
+
+    it("returns false if current path doesn't remain to public API", async () => {
+      ctx.path = "/api/public"
+      expectResult(false)
```
[the remark this answers is not in the document]
Comment (jvcalderon) [reply packages/backend-core/src/utils/tests/utils.spec.ts:185]: Done.

**Pull request merged**